mysql-test/t/percona_mysqldump_innodb_optimize_keys.test

```diff
@@ -392,6 +392,46 @@ INSERT INTO t1(`id`) VALUES (4);
 
 DROP TABLE t1;
 
+#############################################################################
+# Bug PS-4524: Error occurs only if a variable name is a prefix of the
+#              name of the second variable with AUTO_INCREMENT
+#############################################################################
+
+CREATE TABLE `t1` (
+  `prefix` int(5) NOT NULL,
+  `prefix_autoinc` int(5) NOT NULL AUTO_INCREMENT,
+  UNIQUE KEY `prefix_autoinc` (`prefix_autoinc`),
+  PRIMARY KEY (`prefix`)
+) ENGINE=InnoDB;
+
+INSERT INTO t1 VALUES (1,1), (2,2), (3,3);
+INSERT INTO t1(`prefix`) VALUES (4);
+
+CREATE TABLE `t2` (
+  `prefix` int(5) NOT NULL,
+  `prefix_autoinc` int(5) NOT NULL AUTO_INCREMENT,
+  PRIMARY KEY (`prefix`),
+  UNIQUE KEY `prefix_autoinc` (`prefix_autoinc`)
+) ENGINE=InnoDB;
+
+INSERT INTO t2 VALUES (1,10), (2,11), (3,12);
+INSERT INTO t2(`prefix`) VALUES (13);
+
+
+--exec $MYSQL_DUMP --skip-comments --innodb-optimize-keys test t1 t2 >$file
+
+--echo ######################################
+--cat_file $file
```

Comment:
I am not sure we need the dump itself in the output - I'd test after loading the dump SHOW CREATE TABLE instead